src/ShopifyAccess/Services/WebRequestServices.cs

```
@@ -357,65 +224,52 @@ private static string GetLimitFromHeader( HttpHeaders headers )
 			return limitHeader?.FirstOrDefault() ?? string.Empty;
 		}
 
-		private T ParseException< T >( Mark mark, int timeout, Func< T > body )
+		///  <summary>
+		/// 	Handles Shopify server error responses
+		///  </summary>
+		///  <param name="response">Http response</param>
+		///  <param name="mark">Mark</param>
+		///  <exception cref="ShopifyTransientException"></exception>
+		///  <exception cref="ShopifyUnauthorizedException"></exception>
+		///  <exception cref="HttpRequestException"></exception>
+		private async Task ThrowIfErrorAsync( HttpResponseMessage response, Mark mark )
 		{
-			try
-			{
-				return body();
-			}
-			catch( WebException ex )
-			{
-				throw this.LogException( ex, mark );
-			}
-			catch( TaskCanceledException )
-			{
-				ShopifyLogger.LogTimeoutException( mark, this._clientCredentials.ShopName, timeout );
-				throw;
-			}
-		}
+			bool IsTransientHttpStatusCode( HttpStatusCode statusCode ) => statusCode >= HttpStatusCode.InternalServerError || statusCode == HttpStatusCode.RequestTimeout;
 
-		private async Task< T > ParseExceptionAsync< T >( Mark mark, int timeout, Func< Task< T > > body )
-		{
+			var responseStatusCode = response.StatusCode;
+			if( response.IsSuccessStatusCode )
+				return;
+
+			string message;
 			try
 			{
-				return await body();
+				message = await response.Content.ReadAsStringAsync().ConfigureAwait( false );
 			}
-			catch( WebException ex )
+			catch
 			{
-				throw this.LogException( ex, mark );
+				message = "No response message";
```

@SShumer Isn't this actually "Error happened reading content" instead? And we also suppress the exception here. Should've we at least log it and maybe even fail the sync (for example, when we need the response content). Before it seems like we threw in this case

---

@maxim-skuvault 
>Should've we at least log it and maybe even fail the sync
We will fail the sync here in any case (the "response.EnsureSuccessStatusCode();" at the end of this method will throw HttpRequestException as response.IsSuccessStatusCode == false). 
I just want to get an error message if it's available. If not - it's not a problem, we have a status code in any case

---

@SShumer I see what you mean. Execution will get to this point only if it's an unsuccessful response, in which case we'll throw below anyway. But I think it would be clearer if here we did `message = "Unable to read response contents";`

---

Done